components/org.wso2.carbon.identity.oauth.ui/src/main/resources/web/oauth/add-finish-ajaxprocessor.jsp

```diff
@@ -73,50 +74,55 @@
 	OAuthConsumerAppDTO consumerApp = null;
 
 	try {
+        if (OAuthUIUtil.isValidURI(callback)) {
+            String cookie = (String) session.getAttribute(ServerConstants.ADMIN_SERVICE_COOKIE);
```

Comment:
We need to move the java code out of jsp in future, as it is difficult to maintain within JSP.

Reply:
:+1: